### ISSUE TITLE
Composer error "Content-Length Mismatch" again

### ISSUE THREAD

**Issue by p2made** [yiisoft/yii2#17646]
This afternoon Composer started throwing the "Content-Length Mismatch" error on me.

Running `composer diagnose` gives...

```
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: XXXXXXXX XXXXXXXX XXXXXXXX XXXXXXXX  XXXXXXXX XXXXXXXX XXXXXXXX XXXXXXXX
Dev Public Key Fingerprint: XXXXXXXX XXXXXXXX XXXXXXXX XXXXXXXX  XXXXXXXX XXXXXXXX XXXXXXXX XXXXXXXX
OK
Checking composer version: OK
Composer version: 1.9.0
PHP version: 7.3.11
PHP binary path: /usr/local/Cellar/php/7.3.11/bin/php
```

& my Composer `config.json` is...

```
{
    "repositories": {
        "packagist": {
            "type": "composer",
            "url": "https://packagist.org"
        }
    },
    "config": {
        "github-protocols": ["https", "ssh"]
    }
}
```

I'm reporting it here because the default `composer.json` for yii2-advanced is...

```
    "repositories": [
        {
            "type": "composer",
            "url": "https://asset-packagist.org"
        }
    ]
```

& I found that changing that in a couple of projects to...

```
    "repositories": [
        {
            "type": "composer",
            "url": "https://packagist.phpcomposer.com"
        }
    ]
```

resolved it in those.

**Update:** My source for making that change is [this thread](https://github.com/composer/composer/issues/7782) on the Composer repo.

**Comment by samdark** [comment]
Server is out of free space. More storage is being added at the moment. Should be resolved soon.

**Comment by rob006** [comment]
> & I found that changing that in a couple of projects to...
> 
> ```
>     "repositories": [
>         {
>             "type": "composer",
>             "url": "https://packagist.phpcomposer.com"
>         }
>     ]
> ```
> 
> resolved it in those.

This change looks incorrect. https://packagist.phpcomposer.com seems to be some Chinese proxy, you shouldn't use it if you're not from China. I'm also not sure if it contains packages available at https://asset-packagist.org, so this change may affect list of available packages and make some `composer update` calls impossible.

Anyway, this is issue with asset-packagist, you can track progress here: https://github.com/hiqdev/asset-packagist/issues/105

**Comment by p2made** [comment]
**Update:** On later runs Composer sometimes using https://packagist.phpcomposer.com sometimes gave an out of memory error (roughly every other time). Pursing that I pointed Terminal to `~/.composer`, deleted the contents of `cache` & `vendor`, then ran `composer update -v` from that location. That did *not* stop the memory errors, but when I tried https://asset-packagist.org again, the content length mismatch no longer occured. Perhaps work that @samdark mentioned had been completed by then.
